### PR TITLE
FAIR-487 Lat / Long to decimal

### PR DESCRIPTION
**Pull request by ho-henry-senior**
Objectives: 
- to move all location inputs to decimal instead of lat/long
- create a single control to support this that shows airfield chooser then lat long
- to make all location input use new control.
- not to require any datbase schema changes, so: when current process saved port or lat/long to separate fields, new process will too; where current process saves to same, new process will too.

------

- New decimal lat / long macro
- Updated validation of lat long to suit decimal rather than degrees minutes seconds
- Removed unused and obsolete (degrees, minutes, seconds and more) functions from validator.js
- Lat / long to decimal with new macro in:
   - departure location
   - arrival location
   - gar craft location
   - add craft
   - edit craft 
- Unit tests updated
- Moved error and other strings into en.json where found inline.
- Updated airfield dropdown so it’s valid HTML 
- Some updates to manifest page:
   - Colspans corrected 
   - Moved all travel document fields into sub panel
   - Made add person a button 
- Updated airportValidation.js to make includesOneBritishAirport more understandable 
- Moved inline javascript from craft templates into file.

image

### CODE REVIEW

**Comment by ho-henry-senior** [comment]
This is quite a big change, it is probably best if we go through it together. And probably best to be demonstrated. 
I have done quite a lot of testing but will do more.
I might need help with end to end tests.

File: src/app/garfile/arrival/post.controller.js [hunk 1]
```diff
@@ -122,58 +100,30 @@ module.exports = async (req, res) => {
...
-    logger.debug(voyage.arrivalLat);
-    logger.debug(voyage.arrivalLong);
-    const combinedCoordinates = `${voyage.arrivalLat.toString()} ${voyage.arrivalLong.toString()}`;
-    voyage.arrivalPort = combinedCoordinates;
-    logger.debug(voyage.arrivalPort);
-  } else {
-    // If 'Yes' is selected then clear the coordinate values
+
+  if (voyage.portChoice === 'Yes') {
     voyage.arrivalLat = '';
```
Comment (ame-test) [inline src/app/garfile/arrival/post.controller.js:105]: Question: What is the reason why `arrivalLat`/`arrivalLong` is `''` (and in other places is `undefined` or `null`?
*  Note, I will accept their existance.

**Review by ame-test — COMMENTED** [review]
Please answer the below.
* My prefer, since I can't find any obvious bugs, is to have the change merged in and then fix some of the comments in another pull request.
 - I would prefer it deployed on Thursday than not.

**Review by ame-test — APPROVED** [review]
Approve the change, happy for merge in.

File: src/app/garfile/arrival/post.controller.js (hunk 1 above)
Comment (ame-test) [reply src/app/garfile/arrival/post.controller.js:105]: Explanation give, see the below:
```
only one I didn't address was where I dynamically added lat/long fields to Cookie object so could be absent depending on logical route.
I think that is not a problem as no code assumed them to be present and if they are assumed to be a string then coalesces to make sure no error.
```

File: src/common/models/Cookie.class.js [hunk 2]
```diff
@@ -774,14 +717,54 @@ class Cookie {
...
   }
 
+  parseCraftBase(destination) {
+
+    if(!destination.craftBase){
+      destination.portChoice = 'Yes';
+      return;
+    }
+
+    const craftBasePort = destination.craftBase.match(/^[A-Z]{3,4}$/);
```
[the remark this answers is not in the document]
Comment (ame-test) [reply src/common/models/Cookie.class.js]: This has been done.

File: src/common/models/Cookie.class.js [hunk 3]
```diff
@@ -774,14 +717,54 @@ class Cookie {
...
+
+    if (craftBasePort && (craftBaseLat || craftBaseLong)) {
+      throw 'Both port and lat/long supplied for craft base. Supply port or lat/long'
+    }
+
+    if (craftBasePort) {
+      return craftBasePort;
+    }
+
+    return `${craftBaseLat} ${craftBaseLong}`;
```
[the remark this answers is not in the document]
Comment (ame-test) [reply src/common/models/Cookie.class.js:706]: This was not picked up.

**Pull request merged**